_posts/2019-12-12-report-ogp-engine-development.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: "Report: OGP Engine Development"
+tags: ["csc596", "game engine", "simulation", "omega gaming project","report"]
+author: "Gage Coates"
+---
+
+# Omega Gaming Project Overview
+
+The [Omega Gaming Project (OGP)](https://www.omega-gaming-project.org) provides a large number of possible domains to work in, especially those suited for 3D graphics and simulation. As such, the developments in this report vary widely and are in no way representative of a final version of the project.  
+
+## Game
+
+As a game, the player can explore a uniquely different world as often as they like. This world contains a single continent (~4km x 4km) that is populated with dynamically simulated plants, animals, and A.I. controlled NPCs. When a world is created, the initial continent is passed through millions of years of erosion, and gradually populated with evolving plant and animal life. A seed number is used to derive this initial continent, allowing players to start over with the same initial conditions. Once the player interacts with the world, a unique state will emerge that is distinctly unique from any other
+
+## Procedural Content
+
+Due to the ever increasing high budget nature of game development, this project aims to maximize the number of computer generated (procedural) plants, animals, rocks, terrain, textures, quests, names, etc. Content generating algorithms are based on principles that are observed in nature, such as the processes of erosion on terrain, and the gradual adaptation of organisms to their environment through evolution.
+
+## Simulation
+
+Rather than generating the game world at astronomical scales, our approach is to simulate more detail in the confined space of a single continent. Everything from basic newtonian physics to the genetic evolution of plants will be simulated, creating a living world that is always changing. This low level simulation leads to complex emergent patterns that were never explicitly defined. Examples include: the formation of rivers, biomes, plant and animal specialization, cities, road networks, and social networks.
+
+## Artificial Intelligence
+
+Conventional A.I. systems are simple decision trees that require attention to detail and careful consideration of edge cases. These types of A.I. are predictable and can only handle one type of environment. We aim to provide a more general purpose A.I. that interacts with the game and player in a believable and un-planned manner. Since a designer is not explicitly prescribing content, the narrative is primarily driven by each player's experience and decisions.
```

Comment:
```suggestion
Most conventional A.I. systems are simple decision trees that require attention to detail and careful consideration of edge cases. These types of A.I. are predictable and can only handle one set of environmental conditions. We aim to provide a more general purpose A.I. that interacts with the game and player in a believable and unplanned manner. Since a designer is not explicitly prescribing content, the narrative is primarily driven by a player's experience and decisions.
```